File: javascript/lesson 2/middleNode.js
// https://leetcode.com/problems/middle-of-the-linked-list/

function middleNode(head) {
    let slow = head, fast = head;
    while (fast && fast.next) {
        slow = slow.next;
        fast = fast.next.next;
    }
    return slow;
}

File: kotlin/lesson 2/MiddleNode.kt
// https://leetcode.com/problems/middle-of-the-linked-list/

class Solution {
    fun middleNode(head: ListNode?): ListNode? {
        var slow = head; var fast = head
        while (fast?.next != null) {
            slow = slow?.next
            fast = fast.next.next
        }
        return slow
    }
}
